services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -33,58 +37,130 @@ const updateFinalizedHeight = async () => {
 
-	return block;
+	// TODO: Test and fix when indexTransactions is implemented
+	// await indexTransactions(originalBlocks);
 };
 
+const normalizeBlock = block => parseToJSONCompatObj(block);
+
 const getBlocks = async params => {
+	const apiClient = await getApiClient();
+	const blocksDB = await knex('blocks');
 	const blocks = {
 		data: [],
 		meta: {},
 	};
 
-	await Promise.all(
-		['fromTimestamp', 'toTimestamp'].map(async (timestamp) => {
-			if (await validateTimestamp(params[timestamp])) {
-				params[timestamp] = await getBlockchainTime(params[timestamp]);
-			}
-			return Promise.resolve();
-		}),
-	);
+	const { blockId } = params;
+	delete params.blockId;
+	if (blockId) params.id = blockId;
+
+	// TODO: Remove the check when fully implemented. Currently added for cold start bootstrapping.
+	if (!params.heightRange
+		&& !(Object.getOwnPropertyNames(params).length === 1 && params.limit === 1)) {
+		const resultSet = await blocksDB.find(params);
+		if (resultSet.length) params.ids = resultSet.map(row => row.id);
+	}
 
 	const response = await coreApi.getBlocks(params);
-	if (response.data) blocks.data = response.data.map(block => Object
-		.assign(normalizeBlock(block.header), { payload: block.payload }));
+	if (response.data) blocks.data = response.data
+		.map(block => ({ ...block.header, payload: block.payload }));
 	if (response.meta) blocks.meta = response.meta; // TODO: Build meta manually
 
-	blocks.data.map(block => {
+	blocks.data = blocks.data.map(block => {
+		// TODO: Update the below params after accounts index is implemented
+		block.generatorAddress = null;
+		block.generatorUsername = null;
+
 		block.unixTimestamp = block.timestamp;
 		block.totalForged = Number(block.reward);
 		block.totalBurnt = 0;
 		block.totalFee = 0;
 
 		block.payload.forEach(txn => {
-			const txnMinFee = Number(computeMinFee(txn));
+			const txnMinFee = Number(apiClient.transaction.computeMinFee(txn));
 
 			block.totalForged += Number(txn.fee);
 			block.totalBurnt += txnMinFee;
 			block.totalFee += Number(txn.fee) - txnMinFee;
 		});
-		delete block.payload;
-		return block;
+
+		return normalizeBlock(block);
 	});
 
+	indexBlocks(blocks.data);
+
 	return blocks;
 };
 
+const buildIndex = async (from, to) => {
+	logger.info('Building index of blocks');
+
+	if (from >= to) {
+		logger.warn(`Invalid interval of blocks to index: ${from} -> ${to}`);
+		return;
+	}
+
+	const MAX_BLOCKS_LIMIT_PP = 100;
+	const numOfPages = Math.ceil((to + 1) / MAX_BLOCKS_LIMIT_PP - from / MAX_BLOCKS_LIMIT_PP);
+
+	Array(numOfPages).fill().forEach(async (_, pageNum) => {
+		const offset = from + (MAX_BLOCKS_LIMIT_PP * pageNum);
+		logger.info(`Attempting to cache blocks ${offset}-${offset + MAX_BLOCKS_LIMIT_PP}`);
+		// TODO: Revert to standard notation, once getBlocks is fully implemented
+		// const blocks = await getBlocks({
+		// 	limit: MAX_BLOCKS_LIMIT_PP,
+		// 	offset: offset - 1,
+		// 	sort: 'height:asc',
+		// });
+		const blocks = await getBlocks({
```

Comment:
Please consider stand-alone `getBlocksByHeight` in the `coreApi`.

Reply:
This can be done as a part of the issue #244 Retrieve blocks by search index